aspell.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"slices"
+	"strings"
+)
+
+type Aspell []string
+
+func (a Aspell) Check(subject string) error {
+	var words []string
+	var badWords []string
+
+	data, err := checkWithAspellExec(subject)
+	if data != "" {
+		words = strings.Split(data, "\n")
+	}
+	if err != nil {
+		return err
+	}
+
+	for _, word := range words {
+		if len(word) < 1 {
+			continue
+		}
+		if !slices.Contains(a, word) {
+			badWords = append(badWords, word)
+		}
+	}
+
+	if len(badWords) > 0 {
+		return fmt.Errorf("aspell: %s", badWords)
+	}
+	return nil
+}
+
+func (a Aspell) CheckSubjects(subjects []string) error {
+	var response string
+	for _, subject := range subjects {
+		if err := a.Check(subject); err != nil {
+			response += fmt.Sprintf("%s\n", err)
+		}
+	}
+	if len(response) > 0 {
+		return fmt.Errorf("%s", response)
+	}
+	return nil
+}
+
+func LoadAspellExceptions(filename string) (Aspell, error) {
+	var words []string
+
+	var data []byte
+	var err error
+	if data, err = os.ReadFile(filename); err != nil {
+		log.Printf("warning: aspell exceptions file not found (%s)", err)
+	}
+	words = strings.Split(string(data), "\n")
+
+	return words, nil
+}
```

Comment:
Rename function to `NewAspellWithExceptions()` or something with "New".

Reply:
I'll add a package, it is a small app, but regardless its cleaner